### PR TITLE
Prefactorings for `Istio` component

### PR DESCRIPTION
**Pull request by timuthy**
**How to categorize this PR?**

/area control-plane
/kind enhancement
/kind cleanup

**What this PR does / why we need it**:
This PR brings several (p)refactorings to the `Istio` component which is a prerequisite to continue with Istio related tasks in #7016.

Concretely it contains:
- Separation between `Istio-Ingress` and `Istio-System` (outlook: Gardener Operator will manage `Istio-System` and its own `Istio-Ingress`, while Gardenlet will only manage `Istio-Ingress`s if the garden is likewise a seed cluster)
- Makeing `VPN` and `Proxy-Protocol` settings configurable (those are not needed in a garden cluster)
- A fix for deleting outdated stats filters (5710427aae70573b32f8aa2b0dd990f668782259) cc @ScheererJ
- Several cosmetic changes

**Which issue(s) this PR fixes**:
Part of #7016

**Special notes for your reviewer**:
/cc @ScheererJ @rfranzke 

**Release note**:

```bugfix operator
An issues has been fixed that caused outdated Envoy stats filters not being cleaned up in `Istio-Ingress` namespaces.
```

### CODE REVIEW

**Comment by timuthy** [comment]
/test pull-gardener-e2e-kind-operator
see https://github.com/gardener/gardener/issues/7334

File: pkg/operation/botanist/component/istio/istiod.go [hunk 1]
```diff
@@ -57,19 +57,24 @@ var (
...
 	client                    client.Client
 	chartRenderer             chartrenderer.Interface
-	namespace                 string
 	values                    Values
 	istioIngressGatewayValues []IngressGatewayValues
-	istioProxyProtocolValues  []ProxyProtocol
+	istioProxyProtocolValues  []ProxyProtocolValues
+}
+
+// Values contains configuration values for the Istiod component.
```
Comment (ScheererJ) [inline pkg/operation/botanist/component/istio/istiod.go]:
```suggestion
// IstiodValues contains configuration values for the Istiod component.
```

File: pkg/operation/botanist/component/istio/proxy_protocol.go [hunk 2]
```diff
@@ -28,22 +28,17 @@ var (
 )
 
 // ProxyProtocol is a set of configuration values for the istio-proxy-protocol chart.
```
Comment (ScheererJ) [inline pkg/operation/botanist/component/istio/proxy_protocol.go]:
```suggestion
// ProxyProtocolValues is a set of configuration values for the istio-proxy-protocol chart.
```

File: pkg/operation/botanist/component/istio/istiod_test.go [hunk 3]
```diff
@@ -2495,6 +2495,40 @@ spec:
...
+						DNSServerAddress:     pointer.String("1.2.3.4"),
+						NodeLocalIPVSAddress: pointer.String("1.2.3.4"),
+					},
+				)
+			})
+
+			It("should successfully deploy all resources", func() {
+				Expect(c.Get(ctx, client.ObjectKeyFromObject(managedResourceSecret), managedResourceSecret)).To(Succeed())
+				Expect(managedResourceSecret.Type).To(Equal(corev1.SecretTypeOpaque))
+				Expect(managedResourceSecret.Data).To(HaveLen(42))
```
Comment (ScheererJ) [inline pkg/operation/botanist/component/istio/istiod_test.go]: Should the amount of resources not be less than with vpn enabled? I would have expected it to be `40`, i.e. the two resources below should not be present.
WDYT?

File: pkg/operation/botanist/component/istio/istiod_test.go [hunk 4]
```diff
@@ -2529,6 +2521,40 @@ spec:
...
```
Comment (ScheererJ) [inline pkg/operation/botanist/component/istio/istiod_test.go]: Same as with the vpn: I would have expected this to be reduced by the number of proxy protocol specific entities, i.e. I would have expected it to be `39`.
WDYT?

File: pkg/operation/botanist/component/istio/charts/istio/istio-istiod/templates/namespace.yaml [hunk 5]
```diff
@@ -6,4 +6,4 @@ metadata:
   labels:
     istio-operator-managed: Reconcile
     istio-injection: disabled
-{{- end }}
+{{ end -}}
```
Comment (ScheererJ) [inline pkg/operation/botanist/component/istio/charts/istio/istio-istiod/templates/namespace.yaml:14]: Any particular reason for this change? I think that it does not really make a difference here, but we usually use the form that was there before.
WDYT?

File: pkg/operation/botanist/component/istio/istiod_test.go [hunk 6]
```diff
@@ -2555,24 +2563,98 @@ spec:
...
+						DNSServerAddress:     pointer.String("1.2.3.4"),
+						NodeLocalIPVSAddress: pointer.String("1.2.3.4"),
+					},
+				)
+			})
+
+			It("should successfully deploy all resources", func() {
+				Expect(c.Get(ctx, client.ObjectKeyFromObject(managedResourceSecret), managedResourceSecret)).To(Succeed())
+				Expect(managedResourceSecret.Type).To(Equal(corev1.SecretTypeOpaque))
+				Expect(managedResourceSecret.Data).To(HaveLen(27))
```
Comment (ScheererJ) [inline pkg/operation/botanist/component/istio/istiod_test.go]: Here the test is aligned with my expectations!

**Review by ScheererJ — CHANGES_REQUESTED** [review]
Thanks. Looks good overall and is certainly an improvement.
I have a few minor remarks/questions.

File: pkg/operation/botanist/component/istio/istiod.go (hunk 1 above)
Comment (ScheererJ) [reply pkg/operation/botanist/component/istio/istiod.go]: Not relevant anymore as it was fixed with a subsequent commit.

File: pkg/operation/botanist/component/istio/proxy_protocol.go (hunk 2 above)
Comment (ScheererJ) [reply pkg/operation/botanist/component/istio/proxy_protocol.go]: Not relevant anymore as the file was removed in a subsequent commit.

File: pkg/operation/botanist/component/istio/istiod_test.go (hunk 3 above)
Comment (timuthy) [reply pkg/operation/botanist/component/istio/istiod_test.go]: The file itself still exists in the chart but it's empty if VPN is disabled. This is reflected in the `managedSecretSecret` by the registry and therefore I used this [expectation](https://github.com/gardener/gardener/blob/0f36db392f884d7a54b5fced6dba096e130da375/pkg/operation/botanist/component/istio/istiod_test.go#L2528) for example.

File: pkg/operation/botanist/component/istio/charts/istio/istio-istiod/templates/namespace.yaml (hunk 5 above)
Comment (timuthy) [reply pkg/operation/botanist/component/istio/charts/istio/istio-istiod/templates/namespace.yaml:14]: With the latter, the whitespace is really removed which didn't work with `{{- end }}`.

File: pkg/operation/botanist/component/istio/istiod.go (hunk 1 above)
Comment (timuthy) [reply pkg/operation/botanist/component/istio/istiod.go]: The commits are anyway squashed, so I assume we can resolve this?

Comment (ScheererJ) [reply pkg/operation/botanist/component/istio/istiod.go]: Yes, indeed.

File: pkg/operation/botanist/component/istio/istiod_test.go (hunk 3 above)
Comment (ScheererJ) [reply pkg/operation/botanist/component/istio/istiod_test.go]: Ok, that (empty file) is not really nice, but at least I understand now why the number is not as I expected it to be.

File: pkg/operation/botanist/component/istio/charts/istio/istio-istiod/templates/namespace.yaml (hunk 5 above)
Comment (ScheererJ) [reply pkg/operation/botanist/component/istio/charts/istio/istio-istiod/templates/namespace.yaml:14]: Good enough for me.

**Review by ScheererJ — APPROVED** [review]
/lgtm

**Review by rfranzke — COMMENTED** [review]
Well done, very clean, easy to go through the changes.
Thanks!

/approve

**Pull request merged**